mlflow/sklearn/__init__.py

```diff
@@ -925,6 +925,7 @@ def autolog(
     log_post_training_metrics=True,
     serialization_format=SERIALIZATION_FORMAT_CLOUDPICKLE,
     registered_model_name=None,
+    pos_label=None,
```

Comment:
One concern:
```
mlflow.sklearn.autolog(pos_label=X)
```
then it means all sklearn estimator training it must be binary classifier with pos_label=X or regressor (regressor ignore the pos_label), in other cases autologging will fail or generate wrong metric result. @dbczumar Do you think this is a risk ?

Reply:
@WeichenXu123 @wenfeiy-db Yes, this is a concern. We should only make use of `pos_label` when the model is of a relevant type (classifier).

Reply:
@WeichenXu123 I thought for "other cases" (i.e. setting pos_label=X for multi-label question) autolog won't fail, but we'll just have a  warning that it fails to compute the metrics, which is covered by the try-catch [block](https://github.com/mlflow/mlflow/blob/2bf69e5c9a8910d2c1003e128ac4d20ee03fcfce/mlflow/sklearn/utils.py#L489) in `_log_specialized_estimator_content`.  Did I get it wrong?
@dbczumar I believe it's only applied for classifiers right now. Because this PR just passes the `pos_label` to `_log_estimator_content`, which again passes to `_log_specialized_estimator_content`, which only uses the `pos_label` if the model is a classifier ([code link](https://github.com/mlflow/mlflow/blob/f033eae3f95520bcdabbf9bce854fffb63af5c09/mlflow/sklearn/utils.py#L490-L493)). Please correct me if I'm wrong.

Reply:
`pos_label` is ignored for regressors:

https://github.com/mlflow/mlflow/blob/83b778bb4a098e963c1fbb748c93af64ec258fb3/mlflow/sklearn/utils.py#L481-L495

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html#sklearn.metrics.precision_score says:

> pos_labelstr or int, default=1
The class to report if average='binary' and the data is binary. **If the data are multiclass or multilabel, this will be ignored**; setting labels=[pos_label] and average != 'binary' will report scores for that label only.

Reply:
To make it clear, these are different cases can happen when `pos_label` is set and not None:
* If it's a binary classification, autologging uses the given `pos_label` to calculate training metrics and log them
* If it's a multi-label classification, the training metrics calculation will fail (but not the overall autolog). The calculation fails instead of `pos_label` being ignored because we explicitly set `average` to `binary` when having `pos_label` in [_get_classifier_metrics](https://github.com/mlflow/mlflow/blob/816e035786245ff42723b9c53eb9407885e7cd75/mlflow/sklearn/utils.py#L150). In this case autolog just doesn't log any training metrics.
* If it's a regression, pos_label got ignored. The training metrics are calculated as normal.

The above cases sound okay to me. The only doubtful point is that whether we want to log the normal weighted training metrics instead of not logging anything in the 2nd case.

Reply:
@wenfeiy-db Thank you for the clarification. I think that failing the metric computation in the case where `pos_label` is specified for a multi-class classifier is acceptable; can we make sure that we raise a clear error message?

Reply:
Hi @dbczumar I added a test which makes sure that autolog emits warning when the `pos_label` is specified for multi-class. I think the current warning message is pretty clear. Let me know if it looks good.